### PR TITLE
Initial implementation of stream accumulation

### LINKED ISSUE #249: stdout and stderr streams should be accumulated
jupyter notebook:

screenshot 2016-03-22 23 19 25

### PR DESCRIPTION
**Pull request by rgbkrk**
Fixes #249 

Also fixes cases where text needed to be combined in order for escape codes to work properly.

screenshot 2016-03-23 00 15 45

### CODE REVIEW

**Comment by rgbkrk** [comment]
I'm done mucking with this tonight. We can fix it up further later.

**Comment by rgbkrk** [comment]
The way this differs from the Jupyter notebook is that stderr can be buffered before stdout (if it receives stderr first). This made the implementation simpler.

File: src/notebook/agendas/index.js [hunk 1]
```diff
@@ -31,6 +31,35 @@ export function acquireKernelInfo(channels) {
...
+    const nextToLast = outputs.butLast().last();
+    if (nextToLast &&
+        nextToLast.get('output_type') === 'stream' &&
+        nextToLast.get('name') === output.name) {
+      return outputs.updateIn([outputs.size - 2, 'text'], text => text + output.text);
+    }
+  }
+
+  return outputs.push(Immutable.fromJS(output));
+}
```
Comment (jdfreder) [inline src/notebook/agendas/index.js:61]: This implementation looks much tighter than the jupyter/notebook sibling

File: src/notebook/agendas/index.js [hunk 2]
```diff
@@ -70,12 +99,7 @@ export function executeCell(channels, id, source) {
          .ofMessageType(['execute_result', 'display_data', 'stream', 'error', 'clear_output'])
          .map(msgSpecToNotebookFormat)
          // Iteratively reduce on the outputs
-         .scan((outputs, output) => {
-           if (output.output_type === 'clear_output') {
-             return new Immutable.List();
-           }
-           return outputs.push(Immutable.fromJS(output));
-         }, new Immutable.List())
+         .scan(reduceOutputs, emptyOutputs)
```
Comment (jdfreder) [inline src/notebook/agendas/index.js:102]: Slick

**Comment by jdfreder** [comment]
Thanks! :tada:

**Comment by Carreau** [comment]
You guys are up wayyyy to early.

**Pull request merged**